src/snappy_device_agents/devices/multi/__init__.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2023 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Ubuntu multi-device support code."""
+
+import logging
+
+import yaml
+
+import snappy_device_agents
+from snappy_device_agents import logmsg
+from snappy_device_agents.devices import (
+    DefaultDevice,
+)
+from snappy_device_agents.devices.multi.multi import Multi
+from snappy_device_agents.devices.multi.tfclient import TFClient
+
+device_name = "multi"
+
+
+class DeviceAgent(DefaultDevice):
+
+    """Device Agent for provisioning multiple devices at the same time"""
+
+    def provision(self, args):
+        """Method called when the command is invoked."""
+        with open(args.config, encoding="utf-8") as configfile:
+            config = yaml.safe_load(configfile)
+        with open(args.job_data, encoding="utf-8") as jobfile:
+            job_data = yaml.safe_load(jobfile)
```

Comment:
Does jobfile have a YAML format? Iirc, in the previous commit (when you passed args as the argument), you used json to load the data. Maybe I’m wrong, I just want to be sure

Reply:
By the time this file gets to snappy-device-agents, it will be plain json rather than yaml. I don't think it would make a difference since yaml is a superset of json, but I've changed it to use json for reading the file instead.